### PR TITLE
FuzzyMatching works

### LINKED ISSUE #2843: Add option for wildcards in the key of flake registries
One may want to say "overwrite all `github:NixOS/nixpkgs*` to `github:<myName>/nixpkgs*`.
In order to do this, we need to allow one of two options:
1. Add the possibility to use wildcards in the keys of a flake registry
2. Add the possibility to signal that a given map ```github:NixOS/nixpkgs -> github:<myName>/nixpkgs``` is meant to map the base of anything that uses the key as a base.

Syntax:
1. ```"github:NixOS/nixpkgs*": { "uri": "github:<myName>/nixpkgs*" }```
   This would, in the code, match `github:NixOS/nixpkgs/<something>` and convert it to `github:<myName>/nixpkgs/<something>`.
2. ```"github:NixOS/nixpkgs*": { "uri": "github:<myName>/nixpkgs" }``` (without the `*` at the end)
   This would, in the code, match `github:NixOS/nixpkgs/<something>` and convert to `github:<myName>/nixpkgs`.
3. ```"github:NixOS/nixpkgs~":  { "uri": "github:<myName>/nixpkgs" }```
    This would, in the code, match `github:NixOS/nixpkgs/<something>` and convert it to `github:<myName>/nixpkgs/<something>`.

### PR DESCRIPTION
**Pull request by CSVdB**
Fixes #2843

### CODE REVIEW

File: tests/flakes.sh [hunk 1]
```diff
@@ -274,3 +281,51 @@ nix build -o $TEST_ROOT/result --flake-registry $registry flake3:sth
...
+git -C $flake3Dir commit -m 'Remove packages.xyzzy'
+
+# Check that building flake4 still works
+nix build -o $TEST_ROOT/result --flake-registry $registry file://$flake4Dir:xyzzy
+
+# Test whether fuzzy-matching works for IsAlias
+(! nix build -o $TEST_ROOT/result --flake-registry $registry flake4/removeXyzzy:xyzzy)
+
+# Test whether fuzzy-matching works for IsGit
+(! nix build -o $TEST_ROOT/result --flake-registry $registry file://$flake4Dir/removeXyzzy:xyzzy)
```
Comment (edolstra) [inline tests/flakes.sh]: It's probably a good idea to also have a *positive* test for `flake4/removeXyzzy`, e.g. `nix build flake4/removeXyzzy:sth`.

File: tests/config.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  mkDerivation = args:
+    derivation ({
+      inherit system;
+      builder = shell;
+      args = ["-e" args.builder or (builtins.toFile "builder.sh" "if [ -e .attrs.sh ]; then source .attrs.sh; fi; eval \"$buildCommand\"")];
+      PATH = path;
+    } // removeAttrs args ["builder" "meta"])
+    // { meta = args.meta or {}; };
+}
```
Comment (edolstra) [inline tests/config.nix]: I guess you accidentally added this file, can you remove it from the PR?

**Comment by CSVdB** [comment]
Fixed

File: tests/config.nix (hunk 2 above)
Comment (CSVdB) [reply tests/config.nix]: Fixed

**Pull request merged**